### ISSUE TITLE
False-positive of arrow-parens with generic arrow function

### ISSUE THREAD

**Issue by mnn** [gajus/eslint-plugin-flowtype#344]
```js
const t = <A>(a: A): A => {
    const b = a + 1;
    return b;
};
```
I am getting:
> error  Expected parentheses around arrow function argument having a body with curly braces  arrow-parens

which doesn't seem right. Applying fix results in invalid code:

```js
const t = (<)A>(a: A): A => {
```

Workaround is rewrite arrow function to classic function.

**Comment by gajus** [comment]
Fix would be welcome.

**Comment by Mr-Wallet** [comment]
Still getting this problem on `eslint@6.8.0` and `eslint-plugin-flowtype@4.7.0`.

**Comment by gajus** [comment]
Please raise a PR with a broken test case.

**Comment by Mr-Wallet** [comment]
@gajus Question about writing tests: is the assumption that we always be targeting the latest version of flow, and changes to the Flow API are supposed to be handled in future PRs? I don't see any way to target specific versions of flow.

**Comment by gajus** [comment]
> @gajus Question about writing tests: is the assumption that we always be targeting the latest version of flow, and changes to the Flow API are supposed to be handled in future PRs? I don't see any way to target specific versions of flow.

That is correct.

> I'm prepared to leave this for now and move on to other things. But if you happen have any advice for why the tests might have different results than an actual project, I would be eager to try some more experiments to try and nail down why this is happening.

If you can produce a minimal repository for replicating the issue, that would be enough.
